### PR TITLE
Refactor FBP filter padding

### PR DESCRIPTION
**Pull request by askorikov**
This PR is mostly focused on reverting 9cbe3ca and 57982ab (resulting in using a Ram-Lak filter of the same with as the padded sinogram) to improve correspondence to Astra and to follow http://dx.doi.org/10.1109/tns.2014.2363776 more closely. 

Here is the simple test I used:
```python
from ts_algorithms.fbp import ram_lak, filter_sino
import numpy as np
import matplotlib.pyplot as plt


def make_box_phantom():
    x = torch.zeros((64, 64, 64))
    x[8:56, 8:56, 8:56] = 1.0
    x[24:44, 24:44, 24:44] = 0.0
    return x

x = make_box_phantom()
vg = ts.volume(shape=64)
# Use truncated projection data to exacerbate padding importance
pg = ts.cone(angles=96, shape=70, src_det_dist=192)
A = ts.operator(vg, pg)
y = A(x)

# Current implementation
rec_ts = tsa.fdk(A, y, padded=True)

# Zero-padding of the filter
h = ram_lak(y.shape[2])
h = torch.fft.fftshift(h)
h = torch.nn.functional.pad(h, (y.shape[2] // 2, y.shape[2] // 2))
h = torch.fft.ifftshift(h)
rec_zero_padding = tsa.fdk(A, y, padded=True, filter=h)

# Ram-Lak filter of the same width as the padded sinogram
h = ram_lak(2 * y.shape[2])
rec_larger_filt = tsa.fdk(A, y, padded=True, filter=h)

# Astra
rec_astra = astra_fdk(A, y)

# Reconstruction
plot_imgs(
    rec_ts=rec_ts[x.shape[0] // 2],
    rec_zero_padding=rec_zero_padding[x.shape[0] // 2],
    rec_larger_filt=rec_larger_filt[x.shape[0] // 2],
    rec_astra=rec_astra[x.shape[0] // 2]
)
# Differences to the ground truth
plot_imgs(
    rec_ts=(rec_ts - x)[x.shape[0] // 2],
    rec_zero_padding=(rec_zero_padding - x)[x.shape[0] // 2],
    rec_larger_filt=(rec_larger_filt - x)[x.shape[0] // 2],
    rec_astra=(rec_astra - x)[x.shape[0] // 2]
)
# Differences to Astra reconstruction
plot_imgs(
    rec_ts=(rec_ts - rec_astra)[x.shape[0] // 2],
    rec_zero_padding=(rec_zero_padding - rec_astra)[x.shape[0] // 2],
    rec_larger_filt=(rec_larger_filt - rec_astra)[x.shape[0] // 2]
)
# Line profiles
plt.figure()
plt.plot(rec_ts[x.shape[0] // 2, x.shape[1] // 2, :], '--', label='rec_ts')
plt.plot(rec_zero_padding[x.shape[0] // 2, x.shape[1] // 2, :], '--', label='rec_zero_padding')
plt.plot(rec_larger_filt[x.shape[0] // 2, x.shape[1] // 2, :], '--', label='rec_larger_filt')
plt.plot(rec_astra[x.shape[0] // 2, x.shape[1] // 2, :], '--', label='rec_astra')
plt.plot(x[x.shape[0] // 2, x.shape[1] // 2, :], label='phantom')
plt.legend()
``` 
image
image
image
image

As you can see, current implementation and zero-padding of the filter result in worse correspondence to both ground truth and Astra as compared to a wider filter, especially at the edges of the reconstruction. However, I realize I may be missing some rationale for doing the padding this way, so I'm curious about your comments.

### CODE REVIEW

**Comment by D1rk123** [comment]
Thank you for your contribution! I'm sorry for my late reply, but I was on holiday for 4 weeks. I also checked your code without the truncated projections with and without added noise. In those cases, all three approaches perform with a negligible difference.

The fact that there is an improvement when using truncated projections is nice but I don't think it should be a deciding factor in the design of our FBP implementation, because the FBP algorithm was not designed to work with truncated projections. A quick look at the literature suggests that FBP may be used with truncated projections, but typically some kind of sinogram inpainting is done to improve results, which probably makes the filter zero-padding less relevant. If we want good support for truncated projections I would suggest implementing a separate method for this goal in a separate function.

Nevertheless, I think it is an improvement of the code that results from this approach are closer to Astra and that this approach follows the paper that was already cited in the docstring, so I'm in favor of merging this pull request. Good work!
